### ISSUE TITLE
deploying drone.io with tags

### ISSUE THREAD

**Issue by danielepolencic** [dokku/dokku#2235]
**TL;DR:** I'm trying to deploy [drone/drone](https://github.com/drone/drone) in dokku, but I get the following error:

```
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
```

## Description

[drone/drone](https://github.com/drone/drone) is provided as docker image on docker hub. I'm deploying it on dokku as suggested in the documentation:

```shell
dokku apps:create drone
docker pull drone/drone:0.4.2
docker tag drone/drone:0.4.2 dokku/drone:0.4.2
dokku tags:deploy drone 0.4.2
```

the output is:

```
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
-----> Releasing drone (dokku/drone:0.4.2)...
-----> Deploying drone (dokku/drone:0.4.2)...
-----> Attempting to run scripts.dokku.predeploy from app.json (if defined)
-----> No Procfile found in app image
-----> DOKKU_SCALE file found (/home/dokku/drone/DOKKU_SCALE)
=====> web=1
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://dokku.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 10 seconds ...
-----> Default container check successful!
-----> Running post-deploy
-----> Attempting to run scripts.dokku.postdeploy from app.json (if defined)
=====> renaming container (63669ff3c38f) reverent_austin to drone.web.1
-----> Configuring drone.mylocaldokku.dev...(using built-in template)
-----> Creating http nginx.conf
-----> Running nginx-pre-reload
       Reloading nginx
-----> Setting config vars
       DOKKU_APP_RESTORE: 1
-----> Shutting down old containers in 60 seconds
=====> 7fbf252c16fc0932421a6ddf955d5559f4aaf7b8285f1d38b383190d18ddc138
=====> Application deployed:
       http://drone.mylocaldokku.dev:8000
```

After a little bit of investigation I turned on `dokku trace on` and noticed that the container is started withe the following line:

```shell
+ docker run --label=dokku --entrypoint=/bin/sh --rm dokku/drone:0.4.2 -c 'test -f /exec'
docker: Error response from daemon: Container command '/bin/sh' not found or does not exist..
```

I'm not sure where the `/bin/sh` is coming from since the `Dockerfile` in drone has a different `ENTRYPOINT`:

```
FROM centurylink/ca-certs
EXPOSE 8000

ENV DATABASE_DRIVER=sqlite3
ENV DATABASE_CONFIG=/var/lib/drone/drone.sqlite
ENV GODEBUG=netdns=go

ADD release/drone /drone

ENTRYPOINT ["/drone"]
CMD ["server"]
```

## Details

- `uname -a`: Linux vagrant-ubuntu-trusty-64 3.13.0-87-generic #133-Ubuntu SMP Tue May 24 18:32:09 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
- `free -m`: 
```
             total       used       free     shared    buffers     cached
Mem:          2001       1782        219          1        105       1104
-/+ buffers/cache:        571       1429
Swap:            0          0          0
```
- `docker version`:
```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64
```
- `docker -D info`:
```
Containers: 12
 Running: 3
 Paused: 0
 Stopped: 9
Images: 34
Server Version: 1.11.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 150
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 3.13.0-87-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.955 GiB
Name: vagrant-ubuntu-trusty-64
ID: R26R:IRNZ:Q4BK:HZCI:6K3H:UG33:4YMN:BCAF:N7GO:ZBRX:MKSF:3AB5
Docker Root Dir: /var/lib/docker
Debug mode (client): true
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```
- `sigil -v`: 0.4.0
- `docker run -ti gliderlabs/herokuish:latest herokuish version`:
```
herokuish: 0.3.13
buildpacks:
  emberjs.tgz                latest
  buildpack-nginx            v5
  heroku-buildpack-clojure   v75
  heroku-buildpack-erlang    fa17af9
  heroku-buildpack-go        v36
  heroku-buildpack-gradle    v17
  heroku-buildpack-grails    v20
  heroku-buildpack-java      v44
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-nodejs    v90
  heroku-buildpack-php       v102
  heroku-buildpack-play      v26
  heroku-buildpack-python    v80
  heroku-buildpack-ruby      v146
  heroku-buildpack-scala     v70
```
- `dokku version`: 0.5.7
- `dokku plugin`:
```
  00_dokku-standard    0.5.7 enabled    dokku core standard plugin
  20_events            0.5.7 enabled    dokku core events logging plugin
  apps                 0.5.7 enabled    dokku core apps plugin
  build-env            0.5.7 enabled    dokku core build-env plugin
  certs                0.5.7 enabled    dokku core certificate management plugin
  checks               0.5.7 enabled    dokku core checks plugin
  common               0.5.7 enabled    dokku core common plugin
  config               0.5.7 enabled    dokku core config plugin
  docker-options       0.5.7 enabled    dokku core docker-options plugin
  domains              0.5.7 enabled    dokku core domains plugin
  enter                0.5.7 enabled    dokku core enter plugin
  git                  0.5.7 enabled    dokku core git plugin
  logs                 0.5.7 enabled    dokku core logs plugin
  mariadb              1.0.0 enabled    dokku mariadb service plugin
  mongo                1.0.0 enabled    dokku mongo service plugin
  named-containers     0.5.7 enabled    dokku core named containers plugin
  nginx-vhosts         0.5.7 enabled    dokku core nginx-vhosts plugin
  plugin               0.5.7 enabled    dokku core plugin plugin
  proxy                0.5.7 enabled    dokku core proxy plugin
  ps                   0.5.7 enabled    dokku core ps plugin
  rabbitmq             1.0.0 enabled    dokku rabbitmq service plugin
  redis                1.0.0 enabled    dokku redis service plugin
  shell                0.5.7 enabled    dokku core shell plugin
  storage              0.5.7 enabled    dokku core storage plugin
  tags                 0.5.7 enabled    dokku core tags plugin
  tar                  0.5.7 enabled    dokku core tar plugin
```

Environment details (AWS, VirtualBox, physical, etc.): vagrant

How was dokku installed?: `apt-get install -qq -y dokku=0.5.7 herokuish=0.3.13`

**Comment by michaelshobbs** [comment]
That `docker run` command comes from the `is_image_herokuish_based()` function. So in this case it's appropriate that it fails. I think we can just hide the output in that particular function.

**Comment by michaelshobbs** [comment]
My local patched branch looks good

```
$ dokku apps:create drone
Creating drone... done
$ docker pull drone/drone:0.4.2
0.4.2: Pulling from drone/drone
a3ed95caeb02: Pull complete
802d894958a2: Pull complete
a4f9d257f6d0: Pull complete
c3f9f37313f0: Pull complete
Digest: sha256:a630e89873ea687f14414cfe03fafa546abfc2acacb1349a749f2d982cd3a3a3
Status: Downloaded newer image for drone/drone:0.4.2
$ docker tag drone/drone:0.4.2 dokku/drone:0.4.2
$ dokku tags:deploy drone 0.4.2
-----> Setting config vars
       DOKKU_DOCKERFILE_PORTS: 8000/tcp
-----> Releasing drone (dokku/drone:0.4.2)...
-----> Deploying drone (dokku/drone:0.4.2)...
-----> Attempting to run scripts.dokku.predeploy from app.json (if defined)
-----> No Procfile found in app image
-----> DOKKU_SCALE file not found in app image. Generating one based on Procfile...
-----> New DOKKU_SCALE file generated
=====> web=1
-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://dokku.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 10 seconds ...
-----> Default container check successful!
-----> Running post-deploy
-----> Attempting to run scripts.dokku.postdeploy from app.json (if defined)
=====> renaming container (3a0eb0ceee3e) big_lichterman to drone.web.1
-----> Creating new /home/dokku/drone/VHOST...
-----> Configuring drone.dokku.me...(using built-in template)
-----> Creating http nginx.conf
-----> Running nginx-pre-reload
       Reloading nginx
-----> Setting config vars
       DOKKU_APP_RESTORE: 1
=====> Application deployed:
       http://drone.dokku.me:8000
```
